src/test/java/FlightsReporterSpockTests.groovy

```diff
@@ -1,43 +1,197 @@
+        where:
+        flights                                                               | duration | expected
+        new Flights([])                                                       | 30       | []
+        new Flights([shortFlight, mediumLenFlight, longFlight])               | 30       | [shortFlight, mediumLenFlight, longFlight]
+        new Flights([shortFlight, mediumLenFlight, longFlight])               | 80       | [mediumLenFlight, longFlight]
+        new Flights([shortFlight, shortFlight2, mediumLenFlight, longFlight]) | 300      | [longFlight]
+    }
+
+    @Unroll
+    def "get all #flights with #duration"() {
+        expect:
+        FlightsReporter.getAllFlightWithDuration(flights, duration).flights == expected
+
+        where:
+        flights                                                               | duration | expected
+        new Flights([])                                                       | 0        | []
+        new Flights([shortFlight, shortFlight2])                              | 0        | []
+        new Flights([shortFlight, shortFlight2])                              | 60       | [shortFlight, shortFlight2]
+        new Flights([shortFlight, shortFlight2, mediumLenFlight])             | 60       | [shortFlight, shortFlight2]
+        new Flights([shortFlight, shortFlight2, mediumLenFlight, longFlight]) | 120      | [mediumLenFlight]
+    }
+}
+
+class FlightsReporterPassengerTest extends Specification {
+    @Shared
+    Flight empty = FlightGenerator.generateFlight("passengers": 0)
+    @Shared
+    Flight small = FlightGenerator.generateFlight(["passengers": 50])
+    @Shared
+    Flight medium = FlightGenerator.generateFlight(["passengers": 150])
+    @Shared
+    Flight big = FlightGenerator.generateFlight(["passengers": 250])
+
+    def "get flight with most passengers"() {
+        expect:
+        FlightsReporter.getFlightWithMostPassengers(flights) == expected
+
+        where:
+        flights                                  | expected
+        new Flights([])                          | null
+        new Flights([empty])                     | empty
+        new Flights([empty, small, medium, big]) | big
+    }
+
+    def "get flight with least passengers"() {
+        expect:
+        FlightsReporter.getFlightWithLeastPassengers(flights) == expected
+
+        where:
+        flights                                  | expected
+        new Flights([])                          | null
+        new Flights([medium, big])               | medium
+        new Flights([empty, small, medium, big]) | empty
+    }
+
+    def "get #flights with passengers above #count"() {
+        expect:
+        FlightsReporter.getAllFlightsWithPassengersAbove(flights, count).flights == expected
+
+        where:
+        flights                                  | count | expected
+        new Flights([])                          | 0     | []
+        new Flights([empty])                     | -1    | [empty]
```

Comment:
-1  shouldn't be an acceptable value... but anyway
When we do defensive programming https://medium.com/swlh/2-defensive-coding-techniques-you-should-use-today-4225cacc1c29 we'd throw an IllegalArgumentException when we see -1 for example

Reply:
Testing exceptions with Spock is _**fun**_ as well ;)